link_workfow.sh generates unbound variable error on Hera

**Expected behavior**
`./link_workf.sh` should execute without error.

**Current behavior**
`./link_workflow.sh` returns the following error message on Hera
```
[Russ.Treadon@hfe01 sorc]$ ./link_workflow.sh
gfs_utils.fd/ush/detect_machine.sh: line 44: COMPILER: unbound variable
[Russ.Treadon@hfe01 sorc]$
```

**Machines affected**
Error discovered on Hera.   Based on logic in `detect_machine.sh` the error will probably occur on Cheyenne.

**To Reproduce**
1. `git clone https://github.com/NOAA-EMC/global-workflow.git .`
2. `cd sorc`
3. `./checkout.sh -g`
4. `./build_all.sh`
5. `./link_workfow.sh`

Execution of `link_workflow.sh` returns the unbound variable message shown above.

**Additional Information**
Line 44 of `gfs_utils.fd/ush/detect_machine.sh` is the `COMPILER` line shown below
```
# Append compiler (only on machines that have multiple compilers)
if [ $MACHINE_ID = hera ] || [ $MACHINE_ID = cheyenne ]; then
    MACHINE_ID=${MACHINE_ID}.${COMPILER}
fi
```

**Possible Implementation**
Script `sorc/build_all.sh` sets `COMPILER` before executing `detect_machine.sh` as shown below
```
#------------------------------------
# GET MACHINE
#------------------------------------
export COMPILER="intel"
# shellcheck disable=SC1091
source gfs_utils.fd/ush/detect_machine.sh
# shellcheck disable=
if [[ -z "${MACHINE_ID}" ]]; then
  echo "FATAL: Unable to determine target machine"
  exit 1
fi
```
In contrast, `link_workflow.sh` has
```
# shellcheck disable=SC1091
source gfs_utils.fd/ush/detect_machine.sh  # (sets MACHINE_ID)
# shellcheck disable=
machine=$(echo "${MACHINE_ID}" | cut -d. -f1)
```

A possible fix is to add `export COMPILER="intel"` to `link_workflow.sh` as is done in `build_all.sh`.  

A different solution would be to modify `detect_machine.sh` so that it does not reference `COMPILER` for Hera or Cheyenne.

--- a/.gitignore
+++ b/.gitignore
@@ -128,3 +128,8 @@ ush/ufsda
 ush/wafs_blending.sh
 ush/wafs_grib2.regrid.sh
 ush/wafs_intdsk.sh
+ush/finddate.sh
+ush/make_NTC_file.pl
+ush/make_ntc_bull.pl
+ush/make_tif.sh
+ush/month_name.sh

--- a/sorc/link_workflow.sh
+++ b/sorc/link_workflow.sh
@@ -46,6 +46,7 @@ script_dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &> /dev/null && pwd)
 top_dir=$(cd "$(dirname "${script_dir}")" &> /dev/null && pwd)
 cd "${script_dir}"
 
+COMPILER="intel"
 # shellcheck disable=SC1091
 source gfs_utils.fd/ush/detect_machine.sh  # (sets MACHINE_ID)
 # shellcheck disable=

